File: CleanCity/app/src/main/java/com/example/cleancity/modelos/UsuarioModelo.java
package com.example.cleancity.modelos;

import com.example.cleancity.api.Parametro;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.List;

import kotlin.collections.ArrayDeque;

public class UsuarioModelo {
    private String rut;
    private String nombre;
    private String apellido;
    private String sexo;
    private String correo;
    private String sector;
    private String direccion;
    private String telefono;
    private String pass;

    public UsuarioModelo() {
    }

    public UsuarioModelo(String rut, String nombre, String apellido, String sexo, String correo, String sector, String direccion, String telefono, String pass) {
        this.rut = rut;
        this.nombre = nombre;
        this.apellido = apellido;
        this.sexo = sexo;
        this.correo = correo;
        this.sector = sector;
        this.direccion = direccion;
        this.telefono = telefono;
        this.pass = pass;
    }

    public UsuarioModelo(JSONObject fromJSON) throws JSONException {
        this.rut = fromJSON.getString("Rut_User");
        this.nombre = fromJSON.getString("Nombre_User");
        this.apellido = fromJSON.getString("Apellido_User");
        this.sexo = fromJSON.getString("Sexo_User");
        this.correo = fromJSON.getString("Correo_User");
        this.sector = fromJSON.getString("Sector_User");
        this.direccion = fromJSON.getString("Direccion_User");
        this.telefono = fromJSON.getString("Telefono_User");
        this.pass = fromJSON.getString("Pass_User");
    }

    public List<Parametro> busqueda(){
        List<Parametro> retornar = new ArrayDeque<Parametro>();
        try{
            retornar.add(new Parametro("Rut_User", getRut()));
        }catch (Exception e){}

        return retornar;
    }

    public List<Parametro> parametros(){
        List<Parametro> retornar = new ArrayDeque<Parametro>();
        try{
            retornar.add(new Parametro("Rut_User", getRut()));
        }catch (Exception e){}
            retornar.add(new Parametro("Nombre_User", getNombre()));
            retornar.add(new Parametro("Apellido_User", getApellido()));
            retornar.add(new Parametro("Sexo_User", getSexo()));
            retornar.add(new Parametro("Correo_User", getCorreo()));
            retornar.add(new Parametro("Sector_User", getSector()));
            retornar.add(new Parametro("Direccion_User", getDireccion()));
            retornar.add(new Parametro("Telefono_User", getTelefono()));
            retornar.add(new Parametro("Pass_User", getPass()));

        return retornar;
    }

    @Override
    public String toString() {
        return "UsuarioModelo{" +
                "rut='" + rut + '\'' +
                ", nombre='" + nombre + '\'' +
                ", apellido='" + apellido + '\'' +
                ", sexo='" + sexo + '\'' +
                ", correo='" + correo + '\'' +
                ", sector='" + sector + '\'' +
                ", direccion='" + direccion + '\'' +
                ", telefono='" + telefono + '\'' +
                ", pass='" + pass + '\'' +
                '}';
    }

    public String getRut() {
        return rut;
    }

    public void setRut(String rut) {
        this.rut = rut;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public String getSexo() {
        return sexo;
    }

    public void setSexo(String sexo) {
        this.sexo = sexo;
    }

    public String getCorreo() {
        return correo;
    }

    public void setCorreo(String correo) {
        this.correo = correo;
    }

    public String getSector() {
        return sector;
    }

    public void setSector(String sector) {
        this.sector = sector;
    }

    public String getDireccion() {
        return direccion;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }

    public String getPass() {
        return pass;
    }

    public void setPass(String pass) {
        this.pass = pass;
    }
}

File: CleanCityClient/app/src/main/java/com/example/cleancityclient/modelos/UsuarioModelo.kt
package com.example.cleancityclient.modelos

import com.example.cleancityclient.api.Parametro
import org.json.JSONException
import org.json.JSONObject

class UsuarioModelo() {
    private var rut: String? = null
    private var nombre: String? = null
    private var apellido: String? = null
    private var sexo: String? = null
    private var correo: String? = null
    private var sector: String? = null
    private var direccion: String? = null
    private var telefono: String? = null
    private var pass: String? = null

    constructor(rut: String?, nombre: String?, apellido: String?, sexo: String?, correo: String?, sector: String?, direccion: String?, telefono: String?, pass: String?): this() {
        this.rut = rut
        this.nombre = nombre
        this.apellido = apellido
        this.sexo = sexo
        this.correo = correo
        this.sector = sector
        this.direccion = direccion
        this.telefono = telefono
        this.pass = pass
    }

    @Throws(JSONException::class)
    constructor(fromJSON: JSONObject) : this(){
        rut = fromJSON.getString("Rut_User")
        nombre = fromJSON.getString("Nombre_User")
        apellido = fromJSON.getString("Apellido_User")
        sexo = fromJSON.getString("Sexo_User")
        correo = fromJSON.getString("Correo_User")
        sector = fromJSON.getString("Sector_User")
        direccion = fromJSON.getString("Direccion_User")
        telefono = fromJSON.getString("Telefono_User")
        pass = fromJSON.getString("Pass_User")
    }

    fun busqueda(): List<Parametro>? {
        val retornar: MutableList<Parametro> = ArrayDeque<Parametro>()
        try {
            retornar.add(Parametro("Rut_User", getRut()))
        } catch (e: Exception) {
        }
        return retornar
    }

    fun parametros(): List<Parametro>? {
        val retornar: MutableList<Parametro> = ArrayDeque<Parametro>()
        try {
            retornar.add(Parametro("Rut_User", getRut()))
        } catch (e: Exception) {
        }
        retornar.add(Parametro("Nombre_User", getNombre()))
        retornar.add(Parametro("Apellido_User", getApellido()))
        retornar.add(Parametro("Sexo_User", getSexo()))
        retornar.add(Parametro("Correo_User", getCorreo()))
        retornar.add(Parametro("Sector_User", getSector()))
        retornar.add(Parametro("Direccion_User", getDireccion()))
        retornar.add(Parametro("Telefono_User", getTelefono()))
        retornar.add(Parametro("Pass_User", getPass()))
        return retornar
    }

    override fun toString(): String {
        return "UsuarioModelo{" +
                "rut='" + rut + '\'' +
                ", nombre='" + nombre + '\'' +
                ", apellido='" + apellido + '\'' +
                ", sexo='" + sexo + '\'' +
                ", correo='" + correo + '\'' +
                ", sector='" + sector + '\'' +
                ", direccion='" + direccion + '\'' +
                ", telefono='" + telefono + '\'' +
                ", pass='" + pass + '\'' +
                '}'
    }

    fun getRut(): String? {
        return rut
    }

    fun setRut(rut: String?) {
        this.rut = rut
    }

    fun getNombre(): String? {
        return nombre
    }

    fun setNombre(nombre: String?) {
        this.nombre = nombre
    }

    fun getApellido(): String? {
        return apellido
    }

    fun setApellido(apellido: String?) {
        this.apellido = apellido
    }

    fun getSexo(): String? {
        return sexo
    }

    fun setSexo(sexo: String?) {
        this.sexo = sexo
    }

    fun getCorreo(): String? {
        return correo
    }

    fun setCorreo(correo: String?) {
        this.correo = correo
    }

    fun getSector(): String? {
        return sector
    }

    fun setSector(sector: String?) {
        this.sector = sector
    }

    fun getDireccion(): String? {
        return direccion
    }

    fun setDireccion(direccion: String?) {
        this.direccion = direccion
    }

    fun getTelefono(): String? {
        return telefono
    }

    fun setTelefono(telefono: String?) {
        this.telefono = telefono
    }

    fun getPass(): String? {
        return pass
    }

    fun setPass(pass: String?) {
        this.pass = pass
    }
}